### PR TITLE
Cancer biomarkers rewrite

### PR DESCRIPTION
**Pull request by JarrodBaker**

### CODE REVIEW

File: src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala [hunk 1]
```diff
@@ -0,0 +1,168 @@
+package io.opentargets.etl.backend.cancerbiomarkers
+
+import io.opentargets.etl.backend.spark.IoHelpers.IOResources
+import io.opentargets.etl.backend.spark.{IOResource, IOResourceConfig, IoHelpers}
+import io.opentargets.etl.backend.{Configuration, ETLSessionContext}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.LongType
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+
+case class CancerSource(name: String, description: String, link: String)
```
Comment (mkarmona) [inline src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala:10]: @JarrodBaker did you check in the data that there are no empty strings in there? having a description field that is always filled make me wonder whether is something that happens in this reality

File: src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+             struct(
+               col("source") as "name",
+               col("label") as "description",
+               col("url") as "link"
+             ))
+          .otherwise(null)
+      )
+      .drop("source", "label", "url")
+
+    val flatten_distinct = array_distinct _ compose flatten
```
Comment (mkarmona) [inline src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala:126]: well done! Having good practices in mind is always welcome and encouraged.

File: src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    biomarkerDiseaseGeneSources.as[CancerBiomarker]
+  }
+
+  /** Return map on input IOResources */
+  private def getMappedInputs(config: Configuration.CancerBiomarkersSection)(
+      implicit sparkSession: SparkSession): Map[String, IOResource] = {
+
+    val inputs = config.inputs
+
+    val mappedInputs = Map(
```
Comment (mkarmona) [inline src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala:146]: @JarrodBaker you dont need to do this. This is the whole point of having `IOResourceConfig` in the configuration section. You can pass this straight to the map and therefore to the `readFrom` like
```Scala
    val mappedInputs = Map(
      "biomarkers" -> inputs.biomarkers,
      "diseaseMapping" -> inputs.diseaseMapping,
      "sourceMapping" -> inputs.diseaseMapping,
      "targetMapping" -> inputs.targetMapping
    )
    IoHelpers
      .readFrom(mappedInputs)
```

**Review by mkarmona — CHANGES_REQUESTED** [review]

File: src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala (hunk 1 above)
Comment (JarrodBaker) [reply src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala:10]: It's a manually curated file (your favourite) which was previously hidden in the Python source code: all fields exist.

File: src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala (hunk 3 above)
Comment (JarrodBaker) [reply src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala:149]: Thanks for spotting. Fixed!

**Pull request merged**